esphomeRemote/esphomeRemoteLight.h

```diff
@@ -2,30 +2,172 @@
 #include "DisplayUpdateInterface.h"
 #include "MenuTitle.h"
 #include "FriendlyNameEntity.h"
+#include <sstream>
 
 #pragma once
 
+#define MAX_BRIGHTNESS 255.0f
+/* Home Assistant entities can have these modes in color_mode/supported_color_mode.
+ * Depending on the mode the light supports we want to show sliders or not.
+ *
+ * |color_mode | supports brightness | supports color_temp |
+ * |---------- | ------------------- | ------------------- |
+ * | UNKNOWN   |          -          |          -          |
+ * | ONOFF     |          -          |          -          |
+ * | BRIGHTNESS|          x          |          -          |
+ * | COLOR_TEMP|          x          |          x          |
+ * | HS        |          x          |          -          |
+ * | RGB       |          x          |          -          |
+ * | RGBW      |          x          |          -          |
+ * | RGBWW     |          x          |          x          |
+ * | WHITE     |          x          |          -          |
+ * | XY        |          x          |          -          |
+ */
+
 class LightService: public CustomAPIDevice, public Component {
   public:
     LightService(std::string newFriendlyName, std::string newEntityId, DisplayUpdateInterface& newCallback) : friendlyName(newFriendlyName), entityId(newEntityId), display(newCallback) {
       onState = false;
       subscribe_homeassistant_state(&LightService::state_changed, newEntityId.c_str());
+      subscribe_homeassistant_state(&LightService::min_mireds_changed, newEntityId.c_str(),"min_mireds");
+      subscribe_homeassistant_state(&LightService::max_mireds_changed, newEntityId.c_str(),"max_mireds");
+      subscribe_homeassistant_state(&LightService::brightness_changed, newEntityId.c_str(),"brightness");
+      subscribe_homeassistant_state(&LightService::color_temp_changed, newEntityId.c_str(),"color_temp");
+      subscribe_homeassistant_state(&LightService::color_mode_changed, newEntityId.c_str(),"color_mode");
     }
     std::string friendlyName;
     std::string entityId;
     DisplayUpdateInterface& display;
+    int brightness = 0;
+    int color_temp = 0;

```

Comment:
we can make the brightness and colortemp react more smoothly by keeping a localBrightness and localColorTemp. then whenever we scroll the brightness we adjust the local value and send that state. Then if we get a new brightness state that doesn't match our localBrightness we send the localBrightness state again. This is how it works for volume in esphomeRemotePlayer. 

to reduce the number of service calls we can also wait for the brightness state to return from the API or debounce the value to update only once a second

Reply:
Hmm if understand you correctly then this is exactly what it does right now. `brightness` and `color_temp` are local class attributes. When the brightness is increased the class member is incremented and sent to home assistant. When the home assistant value changes the local value is overwritten. This gives me a pretty smooth experience. Did I miss something?

Reducing the number of service calls is a good idea. I changed the implementation to only change a light attribute if the last change has been confirmed by home assistant. I hope further debouncing is not neccessary because I think that would break the smooth changing and we now only send a new request if the last one was accepted.

Reply:
oh after looking at it again that makes perfect sense. I'll play with it a bit more and try to find the cause of my delays

Reply:
Cool but the delays are not cool. What kind of delay are you experiencing? Does the slider update slower then the light?

Reply:
I made a branch called smoothLightDetail that fixes this for me. I only update the local_brightness from the API if it's -1 and then after that it never updates based on the API. this is how volume works

Reply:
Ok yeah I saw how you did it for volume. The reason I chose this was that I wanted the remote brightness be synced if someone changes it from home assistant or wherever.
But if this breaks for you it's bad 😢 could we maybe make this optional? What do you think?

Reply:
I created a new parameter `keep_states_in_sync`. If set to true it will have the behavior I had in mind and similar to volume behavior if set to false. I have to a admit yours is a bit smoother but I like having the slider in sync to home assistant.
Let me know what you think!

Reply:
keep_states_in_sync seems like a good move for now. we can reuse that for the volume stuff. Keeping it off is nice for multiple remotes but it makes sense if you only have 1!